pkg/sharings/settings.go

```diff
@@ -0,0 +1,187 @@
+package sharings
+
+import (
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/instance"
+	"github.com/cozy/cozy-stack/pkg/vfs"
+)
+
+// SharingSettings is the list of destination directories set by the
+// different applications.
+type SharingSettings struct {
+	AppDestination      map[string]map[string]string `json:"app_destination"`
+	SharingsSettingsID  string                       `json:"_id,omitempty"`
+	SharingsSettingsRev string                       `json:"_rev,omitempty"`
+	SharedWithMeDirID   string                       `json:"shared_w_me_dir_id,omitempty"`
+}
+
+// ID returns the SharingSettings qualified identifier.
+func (s SharingSettings) ID() string { return s.SharingsSettingsID }
+
+// Rev returns the SharingSettings revision.
+func (s SharingSettings) Rev() string { return s.SharingsSettingsRev }
+
+// DocType returns the SharingSettings doctype.
+func (s SharingSettings) DocType() string { return consts.Settings }
+
+// Clone returns a new SharingSettings with the same values.
+func (s *SharingSettings) Clone() couchdb.Doc {
+	cloned := *s
+	cloned.AppDestination = make(map[string]map[string]string)
+	for app, dest := range s.AppDestination {
+		appDestMap := make(map[string]string)
+		for doctype, dirID := range dest {
+			appDestMap[doctype] = dirID
+		}
+
+		cloned.AppDestination[app] = appDestMap
+	}
+	return &cloned
+}
+
+// SetID changes the SharingSettings qualified identifier.
+func (s *SharingSettings) SetID(id string) { s.SharingsSettingsID = id }
+
+// SetRev changes the SharingSettings revision.
+func (s *SharingSettings) SetRev(rev string) { s.SharingsSettingsRev = rev }
+
+// Check if the database `consts.Settings` exists, creates it if not, and
+// finally creates the `SharingSettings` document.
```

Comment:
I'm a bit lost here. Why not try to fetch the SharingSettings document, and if it does not exist, create it?